compiler/src/dmd/semantic2.d

```diff
@@ -112,19 +112,39 @@ private extern(C++) final class Semantic2Visitor : Visitor
 
         if (sa.msg)
         {
-            sc = sc.startCTFE();
-            sa.msg = sa.msg.expressionSemantic(sc);
-            sa.msg = resolveProperties(sc, sa.msg);
-            sc = sc.endCTFE();
-            sa.msg = sa.msg.ctfeInterpret();
-            if (StringExp se = sa.msg.toStringExp())
+            OutBuffer msgbuf;
+            for (size_t i = 0; i < sa.msg.length; i++)
             {
-                // same with pragma(msg)
-                const slice = se.toUTF8(sc).peekString();
-                error(sa.loc, "static assert:  \"%.*s\"", cast(int)slice.length, slice.ptr);
+                Expression e = (*sa.msg)[i];
+                sc = sc.startCTFE();
+                e = e.expressionSemantic(sc);
+                e = resolveProperties(sc, e);
+                sc = sc.endCTFE();
+                if (e.type && e.type.ty == Tvoid)
+                {
+                    error(sa.loc, "cannot pass argument `%s` to `static assert` because it is `void`", e.toChars());
```

Comment:
Why? `pragma(msg)` can print types and this artificially disallows void but not other types.

Reply:
Removed. It came from here:
https://github.com/dlang/dmd/pull/11757/files#diff-0ebb96970f91dc91b6b08bcecc0fbd18972450e09bae64df7e87621e09342c79R120